ui/apps/dashboard/src/components/Billing/Addons/EntitlementListItem.tsx

```diff
@@ -92,58 +46,86 @@ export default function EntitlementListItem({
-          entitlement={{
-            currentValue: entitlement.currentValue!, // nil-checked at declaration of selfServiceAvailable above
-            displayValue: entitlement.displayValue,
-            planLimit: entitlement.planLimit!, // nil-checked at declaration of selfServiceAvailable above
-            maxValue: entitlement.maxValue!, // nil-checked at declaration of selfServiceAvailable above
-          }}
-          addon={{
-            price: addon.price!, // nil-checked at declaration of selfServiceAvailable above
-            quantityPer: addon.quantityPer!, // nil-checked at declaration of selfServiceAvailable above
-            addonName: addon.addonName,
-          }}
-          onChange={onChange}
-        />
-      ) : (
-        <div className="flex items-end justify-between">
-          <div>
-            <p className="text-subtle mb-1 flex items-center gap-1 text-sm font-medium">
-              {title}
-              {tooltip && tooltip}
-            </p>
-            <p className="text-muted mb-0.5 text-sm italic">{descriptionText}</p>
-            {entitlement.displayValue && (
-              <p className="text-basis pr-3 text-sm font-medium">{entitlement.displayValue}</p>
-            )}
-          </div>
-          <Button
-            appearance="ghost"
-            label={canIncreaseLimitInCurrentPlan ? 'Contact us' : 'Upgrade'}
-            href={
-              canIncreaseLimitInCurrentPlan
-                ? pathCreator.support({
-                    ref: `app-billing-page-overview-addon-${title
-                      .toLowerCase()
-                      .replace(/ /g, '-')}`,
-                  })
-                : pathCreator.billing({
-                    tab: 'plans',
-                    ref: `app-billing-page-overview-addon-${title
-                      .toLowerCase()
-                      .replace(/ /g, '-')}`,
-                  })
-            }
-          />
+  let content;
+  if (
+    addon &&
+    addon.available &&
+    addon.baseValue !== null &&
+    addon.price !== null &&
+    entitlement.currentValue !== null
+  ) {
+    // The user can increase this entitlement by purchasing an addon.
+
+    content = (
+      <EntitlementListItemSelfService
+        title={title}
+        description={description}
+        tooltip={tooltip}
+        entitlement={{
+          currentValue: entitlement.currentValue,
+          displayValue: entitlement.displayValue,
+          planLimit: addon.baseValue,
+          maxValue: addon.maxValue,
+        }}
+        addon={{
+          price: addon.price,
+          quantityPer: addon.quantityPer,
+          addonName: addon.name,
+        }}
+        onChange={onChange}
+      />
+    );
+  } else {
+    // The user cannot increase this entitlement by purchasing an addon.
+
+    const isCustomPlan = planName.toLowerCase().includes('enterprise');
+    const isHighestPlan = planName === PlanNames.Pro;
+
+    let contactHumanToIncrease = false;
```

Comment:
For some reason I find this name hilarious 😂

Reply:
It's a wonky name but you know exactly what it's for!